src/librustdoc/html/render/mod.rs

```diff
@@ -878,6 +880,7 @@ fn assoc_const(
     indent: usize,
     cx: &Context<'_>,
 ) {
+    let at = AmbiguityTable::empty();
```

Comment:
A lot of `AmbiguityTables` are created way too "low" in the tree call. For example in here, an associated constant is part of an impl or a trait, so the `AmbiguityTable` should have been already created much earlier. I think you can simply create one `AmbiguityTable` in `html/render/context.rs::fn render_item` and then pass it down everywhere.